user-operator/src/main/java/io/strimzi/operator/user/model/KafkaUserModel.java

```diff
@@ -95,6 +96,10 @@ public static KafkaUserModel fromCrd(CertManager certManager,
         result.setAuthentication(kafkaUser.getSpec().getAuthentication());
 
         if (kafkaUser.getSpec().getAuthentication() instanceof KafkaUserTlsClientAuthentication) {
+            if (kafkaUser.getMetadata().getName().length() > 64)    {
```

Comment:
Maybe it could make sense to define a method in the CertManager interface then overloaded by the OpenSslCertManager to return such a value because it is related to openssl and could be different for other tool in the future. But I can live with 64 hard-coded here anyway. It's up to you.